packages/common_client/lib/src/ld_common_client.dart

```diff
@@ -421,39 +422,52 @@ final class LDCommonClient {
       return IdentifyError(Exception(message));
     }
     final res = await _identifyQueue.execute(() async {
-      await _identifyInternal(context,
+      return _identifyInternal(context,
           waitForNetworkResults: waitForNetworkResults);
     });
     return _mapIdentifyResult(res);
   }
 
-  Future<IdentifyResult> _mapIdentifyResult(TaskResult<void> res) async {
+  Future<IdentifyResult> _mapIdentifyResult(
+      TaskResult<IdentifyResult> res) async {
     switch (res) {
-      case TaskComplete<void>():
-        return IdentifyComplete();
-      case TaskShed<void>():
+      case TaskComplete<IdentifyResult>(result: var result):
+        return result ?? IdentifyComplete();
+      case TaskShed<IdentifyResult>():
         return IdentifySuperseded();
-      case TaskError<void>(error: var error):
+      case TaskError<IdentifyResult>(error: var error):
         return IdentifyError(error);
     }
   }
 
-  Future<void> _identifyInternal(LDContext context,
+  Future<IdentifyResult> _identifyInternal(LDContext context,
       {bool waitForNetworkResults = false}) async {
-    await _setAndDecorateContext(context);
-    final completer = Completer<void>();
-    _eventProcessor?.processIdentifyEvent(IdentifyEvent(context: _context));
-    final loadedFromCache = await _flagManager.loadCached(_context);
-
-    if (_config.offline) {
-      return;
+    if (!context.valid) {
+      const message =
+          'LDClient was provided an invalid context. The context will be ignored. Existing flags will be used for evaluations until identify is called with a valid context.';
+      _logger.warn(message);
+      return IdentifyError(Exception(message));
```

Comment:
I think we should probably use a custom exception type. Like `InvalidContextException`.